driver-core/src/main/com/mongodb/internal/connection/BsonWriterAdapter.java

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.connection;
+
+import org.bson.BsonBinary;
+import org.bson.BsonDbPointer;
+import org.bson.BsonReader;
+import org.bson.BsonRegularExpression;
+import org.bson.BsonTimestamp;
+import org.bson.BsonWriter;
+import org.bson.types.Decimal128;
+import org.bson.types.ObjectId;
+
+import static org.bson.assertions.Assertions.notNull;
+
+public class BsonWriterAdapter implements BsonWriter {
+    private final BsonWriter bsonBinaryWriter;
```

Comment:
Using `bsonBinaryWriter` as the name is confusing as it might not be a `BsonBinaryWriter` instance.  
Also regarding naming is this an Adapter class or just a Wrapper class?

Reply:
* Yes, changed the field name.
* Yes, it's definitely not an Adapter (https://en.wikipedia.org/wiki/Adapter_pattern).  I think the common term is Decorator (https://en.wikipedia.org/wiki/Decorator_pattern) so changed to that suffix.